features/step_definitions/credit_card_steps.rb

```diff
@@ -112,3 +112,7 @@
     find(:css, '#braintree_nonce', visible: :hidden).set('some_braintree_nonce')
   end
 end
+
+When "Stripe is going to fail" do
+  PaymentGateways::StripeCrypt.any_instance.stubs(:create_stripe_setup_intent).returns(nil)
```

Comment:
Ultimately, this method reduces to https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/stripe_response.rb#L74. I see it very hard that it would return `nil`. This requires more investigation.

Reply:
Well I think it's the only explanation for `@intent` to be `nil` 🤔

Reply:
I think @josemigallas is right it could come from https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/stripe_client.rb#L558

Either stripe did not respond or the response was ununderstandable

Reply:
The line you pointed goes in a `raise`, @hallelujah. Where are rescuing from it?

Reply:
1. https://github.com/3scale/porta/blob/ca33031586eda06b77823e2c03b9b56d7ee3937f/lib/developer_portal/app/controllers/developer_portal/admin/account/stripe_controller.rb#L8

2. https://github.com/3scale/porta/blob/ca33031586eda06b77823e2c03b9b56d7ee3937f/app/lib/payment_gateways/stripe_crypt.rb#L34

3. https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/resources/setup_intent.rb#L6

4. https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/api_operations/create.rb#L7

    4.1. https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/api_operations/request.rb#L25

    4.2. https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/stripe_client.rb#L241

    4.3. https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/stripe_response.rb#L74

5. https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/api_operations/create.rb#L8

    5.1. https://github.com/stripe/stripe-ruby/blob/9787913b355049af915cf075b29666e55c5ab7f2/lib/stripe/util.rb#L74